src/cascadia/WindowsTerminal/WindowUiaProvider.cpp

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+#include "pch.h"
+
+#include "WindowUiaProvider.hpp"
+#include "../types/ScreenInfoUiaProvider.h"
+
+#include "../host/renderData.hpp"
+
+WindowUiaProvider::WindowUiaProvider(Microsoft::Console::Types::IUiaWindow* baseWindow) :
+    //_pScreenInfoProvider{ nullptr },
+    WindowUiaProviderBase(baseWindow)
+{
+}
+
+WindowUiaProvider::~WindowUiaProvider()
+{
+    /*if (_pScreenInfoProvider)
+    {
+        _pScreenInfoProvider->Release();
+    }*/
+}
+
+WindowUiaProvider* WindowUiaProvider::Create(Microsoft::Console::Types::IUiaWindow* baseWindow)
+{
+    WindowUiaProvider* pWindowProvider = nullptr;
+    Microsoft::Console::Types::ScreenInfoUiaProvider* pScreenInfoProvider = nullptr;
+    try
+    {
+        pWindowProvider = new WindowUiaProvider(baseWindow);
+
+        // TODO GitHub #1352: Hook up ScreenInfoUiaProvider to WindowUiaProvider
+        /*Globals& g = ServiceLocator::LocateGlobals();
+        CONSOLE_INFORMATION& gci = g.getConsoleInformation();
+        Microsoft::Console::Render::IRenderData* renderData = &gci.renderData;
+
+        pScreenInfoProvider = new Microsoft::Console::Types::ScreenInfoUiaProvider(renderData, pWindowProvider);
+        pWindowProvider->_pScreenInfoProvider = pScreenInfoProvider;
+        */
+
+        // TODO GitHub #1914: Re-attach Tracing to UIA Tree
+        //Tracing::s_TraceUia(pWindowProvider, ApiCall::Create, nullptr);
+
+        return pWindowProvider;
+    }
+    catch (...)
+    {
+        if (nullptr != pWindowProvider)

```

Comment:
This part was copied over from the ConHost code when I created the inheritance model with `WindowUiaProviderBase`. Since we've been talking about switching over to `wil` in this area and the `WindowUiaProvider` is going to be important for changing the navigation of the tree (one of the upcoming tasks), could I just add a work item here and call it a day?

Reply:
just so long as we do get to it.